File: atcoder/abc/abc150/e/main.py
N = int(input())
C = [int(i) for i in input().split()]
mod = 10**9+7
rev = pow(2, mod-2, mod)
C.sort()
ans = 0
for i in range(N):
  ans += C[i]*(1+(N-i))*rev%mod*pow(2, 2*N-1, mod)
  ans %= mod
print(ans)

File: atcoder/abc/abc150/e/main.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long
const ll mod = 1e9+7;


ll pow(ll x, ll y){
  ll ret = 1;
  while(y>0){
    if(y&1) ret = ret*x%mod;
    x = x*x%mod;
    y >>= 1;
  }
  return ret;
}

int main(){
  int N;
  cin >> N;
  vector<ll> C(N);
  for(int i=0;i<N;i++){
    cin >> C[i];
  }
  ll rev = pow(2, mod-2);
  cout << rev << endl;
  sort(C.begin(), C.end());
  ll ans = 0;
  for(int i=0;i<N;i++){
    ans += C[i]*(N-i+1)%mod*pow(2ll, (ll)(2*N-1))%mod*rev%mod;
  }
  cout << ans << endl;
}
